### DISCUSSION TITLE
First Thoughts

### DISCUSSION THREAD

**Discussion by alibey-10** [Ali-Hussein-dev/Q-Lib#1] [General]
Al Salam Alaykum everyone.

I want to share with you some thoughts, I am thinking of building a library around Quran. It would look like a documentation site but with books. The motivation is to help Muslims understand the Quran with help of the **most valuable books** written about Quran like Tafseer or similar effortlessly. I personally find it very useful, a time saver, and worthy of having a such thing because there is a lot of info outside but it is not easy to find good resources. 

## Initial Features for the first version
- simple user (reader) preferences that will be saved on the browser. (such font size, dark mode, etc.)
- PWA 
- Only Arabic books will be included

## Future Features
- search box
- open books from the last page
- bookmarks
- rate book 
- recommend a book
- donate to the author
- etc

## Stack
- Next.js (But I am keeping an eye on Astro)
- Sanity 
- Database (SQL or NoSql) (still don't know)
- MDX
- Tailwindcss
- Mantine

What do you think about the IDEA and about the technical challenges?
Is anyone interested in the idea?
Please share your thoughts and experiences with me.

I will update this repo with my progress and the next step will be a demo to see what it is going to look like.

**Comment by AyahBot-Dev** [comment]
One suggestion, try to avoid JS Frameworks with Virtual DOMs as much as possible.
